### PR TITLE
Throwing out the welcome mat

### PR DESCRIPTION
**Pull request by ActuallyTaylor**
Removed unnecessary code that caused the heap to overflow and the board to crash when too many objects were initialized. Also fixed the crash that would happen when the HUDL received a crashed status. Most of these fixes were solved by removing 1024 byte EVT logo...

Signed-off-by: ActuallyTaylor <taylor.lineman@gmail.com>

### CODE REVIEW

File: src/HUDL/HUDL.cpp [hunk 1]
```diff
@@ -41,62 +36,49 @@ uint16_t HUDL::getObjectDictionarySize() const {
...
-            lcd.clearLCD();
-            headerForCorner(TOP_LEFT, "Battery");
-            headerForCorner(TOP_RIGHT, "HI Temp");
-            headerForCorner(BOTTOM_LEFT, "RPM");
-            headerForCorner(BOTTOM_RIGHT, "MC Stat");
-            setHeaders = true;
-        }
+    if (!setHeaders) {
+        lcd.clearLCD();
+        headerForCorner(TOP_LEFT, "Battery");
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]: This isn't really part of the goal of this PR, but can you change this, so it doesn't say "Battery HI Temp" anymore. The riders were getting confused by it

File: src/HUDL/HUDL.cpp [hunk 2]
```diff
@@ -41,62 +36,49 @@ uint16_t HUDL::getObjectDictionarySize() const {
...
+    char temp[9];
+    std::sprintf(temp, "0x%02X C", highestTemp);
+    dataForCorner(TOP_RIGHT, temp);
+
+    // Set the rpm
+    char rpm[8];
+    std::sprintf(rpm, "%u", actualPosition);
+    dataForCorner(BOTTOM_LEFT, rpm);
+
+    // Set the status word
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]:
```suggestion
    // Set the motor controller status word
```

File: src/HUDL/HUDL.cpp [hunk 3]
```diff
@@ -113,7 +95,7 @@ void HUDL::headerForCorner(HUDL::Corner corner, const char* text) {
     sectionColumn += padding;
 
     // Write the text to the screen under the section header.
-    lcd.writeLargeText(text, sectionPage, sectionColumn, false);
+    lcd.writeText(text, sectionPage, sectionColumn, EVT::core::DEV::LCD::FontSize::LARGE, false);
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp:98]: We need to find a solution for really long enums at some point, but probably not right now

File: src/HUDL/HUDL.cpp [hunk 4]
```diff
@@ -130,21 +112,12 @@ void HUDL::dataForCorner(HUDL::Corner corner, const char* text) {
...
-
-    // Write the Error header
-    lcd.writeLargeText("!!! Error !!!", 0, padding, false);
-
-    // Write the error message
-    lcd.writeSmallText(text, 2, 0, true);
+    lcd.writeText(text, sectionPage, sectionColumn, EVT::core::DEV::LCD::FontSize::LARGE, false);
 }
 
+// Ignore the -Wreturn-type warning that occurs here because of the switch statements.
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]: What is this issue? If it's yelling at you for not having a default case, you should probably just add a default case

File: targets/HUDL/main.cpp [hunk 5]
```diff
@@ -21,12 +21,12 @@ namespace time = EVT::core::time;
...
 
-const uint32_t SPI_SPEED = SPI_SPEED_500KHZ;
-const uint8_t deviceCount = 1;
+#define SPI_SPEED SPI_SPEED_500KHZ
+#define DEVICE_COUNT 1
 
 ///////////////////////////////////////////////////////////////////////////////
-// EVT-core CAN callback and CAN setup. This will include logic to set
-// aside CANopen messages into a specific queue
+// EVT-core CAN call back and CAN set up. This will include logic to set     //
```
Comment (mjmagee991) [inline targets/HUDL/main.cpp]: Are callback and setup not valid single words? Why is there a `//` at the end of this line?

File: targets/HUDL/main.cpp [hunk 6]
```diff
@@ -225,7 +168,34 @@ int main() {
         COTmrService(&canNode.Tmr);
         // Handle executing timer events that have elapsed
         COTmrProcess(&canNode.Tmr);
-        // Wait for new data to come in
-        //        time::wait(10);
     }
 }
+
```
Comment (mjmagee991) [inline targets/HUDL/main.cpp]: Why did you move these down here? I see why you might argue this makes the code a bit cleaner, but if you want to modify how we handle this, we should change every board to match. If this is something you want, let's talk more

File: targets/HUDL/main.cpp [hunk 7]
```diff
@@ -21,12 +21,12 @@ namespace time = EVT::core::time;
...
-const uint32_t SPI_SPEED = SPI_SPEED_500KHZ;
-const uint8_t deviceCount = 1;
+#define SPI_SPEED SPI_SPEED_500KHZ
+#define DEVICE_COUNT 1
 
 ///////////////////////////////////////////////////////////////////////////////
-// EVT-core CAN callback and CAN setup. This will include logic to set
-// aside CANopen messages into a specific queue
+// EVT-core CAN call back and CAN set up. This will include logic to set     //
+// aside CANOpen messages into a specific queue                              //
```
Comment (mjmagee991) [inline targets/HUDL/main.cpp]: [CANopen](https://en.wikipedia.org/wiki/CANopen)
```suggestion
// aside CANopen messages into a specific queue                              //
```

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Looks pretty good, but formatting is a bit strange

File: src/HUDL/HUDL.cpp (hunk 3 above)
Comment (ActuallyTaylor) [reply src/HUDL/HUDL.cpp:98]: Yeah maybe we could do a namespace thing? I see a lot of libraries using macros but I'm not sure if that is very intuitive or expressive though.

File: src/HUDL/HUDL.cpp (hunk 4 above)
Comment (ActuallyTaylor) [reply src/HUDL/HUDL.cpp]: It is yelling because two of the cases are "missing". They are just combined into one line for formatting. A warning here is thrown because it thinks two cases are missing, but they aren't.

File: targets/HUDL/main.cpp (hunk 5 above)
Comment (ActuallyTaylor) [reply targets/HUDL/main.cpp]: Oops that is just a spelling mistake

File: targets/HUDL/main.cpp (hunk 6 above)
Comment (ActuallyTaylor) [reply targets/HUDL/main.cpp]: I think I moved them down here because I was doing a lot of up and down scrolling in a main file and just kept getting lost in the weeds there. I do want to talk about that though because I am really not sure what is better for them. They kind of disappear when they are at the bottom, which has once led me to think they were not in there at all. I think this can be addressed in the CANOpen upgrade, so we should talk about this.

File: src/HUDL/HUDL.cpp (hunk 4 above)
Comment (mjmagee991) [reply src/HUDL/HUDL.cpp]: When I changed the code to the below, the warning went away:
```
    switch (corner) {
    case TOP_LEFT:
    case TOP_RIGHT:
        return 0;
    case BOTTOM_LEFT:
    case BOTTOM_RIGHT:
        return 4;
    default:
        return -1;
    }
```
However, when I changed the code to this, the error was still there:
```
    switch (corner) {
    case TOP_LEFT:
        return 0;
    case TOP_RIGHT:
        return 0;
    case BOTTOM_LEFT:
        return 4;
    case BOTTOM_RIGHT:
        return 4;
    }
```
If I'm misunderstanding what you mean, let me know. I'm okay with disabling this warning if it makes sense to, but I'd like to avoid it if at all possible.

File: src/HUDL/HUDL.cpp (hunk 1 above)
Comment (mjmagee991) [reply src/HUDL/HUDL.cpp]: I saw you change this, but it doesn't look like those changes got pushed

File: targets/HUDL/main.cpp [hunk 8]
```diff
@@ -42,60 +42,21 @@ const uint8_t deviceCount = 1;
...
-    log::LOGGER.log(log::Logger::LogLevel::DEBUG, "RX Full");
-}
-
 int main() {
     // Initialize system
     EVT::core::platform::init();
 
-    // Will store CANopen messages that will be populated by the EVT-core CAN
+    IO::UART& uart = IO::getUART<IO::Pin::PB_9, IO::Pin::PB_8>(9600);
+    log::LOGGER.setUART(&uart);
```
Comment (mjmagee991) [inline targets/HUDL/main.cpp:81]: You should also set the log level here. It isn't explicitly set to a default value in the Logging class (which is more of an issue with the logger itself, but that's a problem for another day...)

File: targets/HUDL/main.cpp (hunk 6 above)
Comment (mjmagee991) [reply targets/HUDL/main.cpp]: Grab me on Monday, so we can discuss this

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Almost ready to go

File: src/HUDL/HUDL.cpp [hunk 9]
```diff
@@ -175,6 +136,10 @@ uint8_t HUDL::pageForCorner(HUDL::Corner corner) {
     case BOTTOM_LEFT:
     case BOTTOM_RIGHT:
         return 4;
+    default:
+        return -1;
     }
 }
+#pragma GCC diagnostic pop
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]:
```suggestion
```

File: src/HUDL/HUDL.cpp [hunk 10]
```diff
@@ -41,62 +36,49 @@ uint16_t HUDL::getObjectDictionarySize() const {
...
-        break;
     }
+
+    char temp[9];
+    std::sprintf(temp, "%hu.%hu C", highestTemp / 10, highestTemp % 10);
+    dataForCorner(TOP_RIGHT, temp);
+
+    // Set the rpm
+    char rpm[8];
+    std::sprintf(rpm, "%u", actualPosition);
```
Comment (mjmagee991) [inline src/HUDL/HUDL.cpp]: I've been getting complaints from integration about RPM rolling over to negatives when they spin the motor backwards. It's not really related to this PR, but can you change this here and change the data type of `actualPosition`?
```suggestion
    std::sprintf(rpm, "%d", actualPosition);
```

File: targets/HUDL/main.cpp [hunk 11]
```diff
@@ -225,7 +193,5 @@ int main() {
         COTmrService(&canNode.Tmr);
         // Handle executing timer events that have elapsed
         COTmrProcess(&canNode.Tmr);
-        // Wait for new data to come in
-        //        time::wait(10);
     }
-}
+}
```
Comment (mjmagee991) [inline targets/HUDL/main.cpp]:
```suggestion
}

```

**Review by mjmagee991 — APPROVED** [review]
Three minor details, and then you're good to merge

**Review by mjmagee991 — APPROVED** [review]

File: targets/HUDL/main.cpp [hunk 12]
```diff
@@ -107,36 +93,20 @@ int main() {
     DEV::Timerf3xx timer(TIM2, 160);
 
     //create the RPDO node
-    IO::GPIO* devices[deviceCount];
+    IO::GPIO* devices[DEVICE_COUNT];
 
     IO::GPIO& regSelect = IO::getGPIO<IO::Pin::PA_3>(EVT::core::IO::GPIO::Direction::OUTPUT);
```
Comment (mjh9585) [inline targets/HUDL/main.cpp]:
```suggestion
    IO::GPIO& regSelect = IO::getGPIO<IO::Pin::PA_3>(IO::GPIO::Direction::OUTPUT);
```
Is the EVT::core necessary on these lines since the IO namespace is defined above.

**Review by mjh9585 — APPROVED** [review]
Just one small change, otherwise good.
